public/app/core/components/AppChrome/TopBar/TopBarExtensionPoint.tsx

```diff
@@ -0,0 +1,28 @@
+import { renderLimitedComponents } from '@grafana/runtime';
+import { usePluginComponents } from 'app/features/plugins/extensions/usePluginComponents';
+
+/**
+ * Extension point for plugins to add components to the top bar.
+ * Currently restricted to grafana-setupguide-app plugin.
+ */
+export function TopBarExtensionPoint() {
+  const { components } = usePluginComponents({
+    extensionPointId: 'grafana/singletopbar/action',
+  });
+
+  // Return null if no components are registered
+  if (components.length === 0) {
+    return null;
+  }
+
+  return (
+    <>
+      {renderLimitedComponents({
```

Comment:
Total nitpick: Is there a reason we can't return the method here instead of wrapping in a fragment? (And the same for the other extension point)

```tsx
return renderLimitedComponents({ ...
```